### ISSUE TITLE
auto validate record counts

### ISSUE THREAD

**Issue by patrickzurek** [eXtensibleCatalog/xcmetadataservicestoolkit#757]
JIRA issue created by: rcook
Originally opened: 2011-03-03 05:06 PM

Issue body:
 (nt)

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2011-05-11 06:07 PM

Comment body:

Per decision previously made by Dave, he did not want a stats page to be built.   I still am struggling with how to validate numbers and find opening all the different logs and then hopping back and forth between them to be frustrating.

I am moving this as assigned to Dave for future consideration/discussion.

**Comment by patrickzurek** [comment]
JIRA Coment by user: dlindahl
JIRA Timestamp: 2011-06-08 02:45 PM

Comment body:

We should capture a description of the manual tests that Randy wants to be able to do, so that they can be coded and done by the software, and the results can be output to the log files as OK or ERROR lines.

**Comment by patrickzurek** [comment]
JIRA Coment by user: banderson
JIRA Timestamp: 2011-06-10 01:37 PM

Comment body:

as per discussion with Randy and John, we all agreed this would be nice.  Not only for users, but for us because manually verifying these numbers is grueling.  We already have all the data we need.  We just need to put the rules in place.  Here's what we came up with.

1) rules should be configurable so that other institutions can either change them per their particular data set or just turn them off.
2) add hooks into services to run the rules
3) spew output to one MST-wide rules output log
4) add that log to the "General Logs"screen

and here's our initial set of rules:

H=harvest, N=Normalization, T=Transformation
A=active, D=deleted, H=held
T=total, B=bib, H=holding, W=work, E=expression, M=manifestation, H=holding

HBA  =  NBA
HHA  =  NHA
HTA  =  NTA

NBA  =  TMA
NBD  =  TMD - I'm not positive about this one.
NHA  =  THA+THH
NHD  =  THD - again, I'm not sure

TWA >=  TMA
TEA >=  TMA
TEA  =  TWA

we discussed these in a meeting, but I realized afterwards that they are not true:

HBD  =  NBD
HHD  =  NHD
HTD  =  NTD

The reason these won't be true is that the norm service doesn't create new records for incoming deleted records that it hasn't seen before.  There would really be no point in doing so.
